util/push/push.go

```diff
@@ -138,6 +138,17 @@ func annotateDistributionSourceHandler(manager content.Manager, f images.Handler
 
 		for i := range children {
 			child := children[i]
+
+			if m, ok := annotations[child.Digest]; ok {
+				for k, v := range m {
+					if child.Annotations == nil {
+						child.Annotations = map[string]string{}
+					}
+					child.Annotations[k] = v
+				}
+			}
+			children[i] = child
```

Comment:
nit: Don't think it makes a functional difference at the moment but this func will now add all annotations rather than just `containerd.io/distribution.source.*`, which makes the name `annotateDistributionSourceHandler` misleading. Maybe it should change names or maybe this loop should be updated to skip non-`distribution.source` annotations.

Reply:
updated